lib/src/diff.rs

```diff
@@ -587,44 +587,24 @@ impl<'input> Diff<'input> {
     }
 }
 
-#[derive(PartialEq, Eq, Clone)]
+#[derive(PartialEq, Eq, Clone, Debug)]
 pub enum DiffHunk<'input> {
-    Matching(&'input [u8]),
-    Different(Vec<&'input [u8]>),
+    Matching(&'input BStr),
+    Different(Vec<&'input BStr>),
 }
 
 impl<'input> DiffHunk<'input> {
-    pub fn matching<T: AsRef<[u8]> + ?Sized>(content: &'input T) -> Self {
+    pub fn matching<T: AsRef<BStr> + ?Sized>(content: &'input T) -> Self {
         DiffHunk::Matching(content.as_ref())
     }
 
-    pub fn different<T: AsRef<[u8]> + ?Sized + 'input>(
+    pub fn different<T: AsRef<BStr> + ?Sized + 'input>(
         contents: impl IntoIterator<Item = &'input T>,
     ) -> Self {
         DiffHunk::Different(contents.into_iter().map(AsRef::as_ref).collect())
     }
 }
 
-impl Debug for DiffHunk<'_> {
```

Comment:
Are there any tests for this? I assumed there were, but I didn't immediately find them.

Reply:
I don't think there are, but it might affect assertion failure.

Reply:
(Optional) do you think it would make sense to add one?

Reply:
Maybe no? It's not so important, and the point of deriving `Debug` is that the generated code is less buggy.

Reply:
Sounds good.